server/py/framework/db/sqldb/db.py

```diff
@@ -589,6 +580,23 @@ def _ensure_run_name_on_update(run_record: Run, run_dict: dict):
                 "Changing name for an existing run is invalid"
             )
 
+    @staticmethod
+    def _update_run_end_time(run: Run, run_dict: dict, end_time: Optional[str] = None):
```

Comment:
Please add a comment here that explains what is the case where we don't want to update the run's end time, cause the if statements here looks a bit confusing at first sight